From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
Date: Mon, 26 Mar 2018 18:30:53 +0300
Subject: [PATCH] Upstream config migrations

This patch contains config migrations for when upstream adds options
which Paper already had.

diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 8007220fa0646414c9cec66161aca2394e62dde3..6b952280399833eec54a72e87d916e65ae4c7339 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -288,4 +288,22 @@ public class PaperConfig {
     private static void authenticationServersDownKickMessage() {
         authenticationServersDownKickMessage = Strings.emptyToNull(getString("messages.kick.authentication-servers-down", authenticationServersDownKickMessage));
     }
+
+    private static void savePlayerData() {
+        Object val = config.get("settings.save-player-data");
+        if (val instanceof Boolean) {
+            SpigotConfig.disablePlayerDataSaving = !(Boolean) val;
+            SpigotConfig.config.set("players.disable-saving", SpigotConfig.disableAdvancementSaving);
+            SpigotConfig.save();
+        }
+    }
+
+    private static void namedEntityDeaths() {
+        Object val = config.get("settings.log-named-entity-deaths");
+        if (val instanceof Boolean bool && !bool) {
+            SpigotConfig.logNamedDeaths = false;
+            SpigotConfig.config.set("settings.log-named-deaths", false);
+            SpigotConfig.save();
+        }
+    }
 }
